torch/_dynamo/exc.py

```diff
@@ -212,7 +212,7 @@ def filter_stack(stack):
     for frame in stack:
         if "convert_frame" in frame.filename:
             break
-        if "eval_frame" in frame.filename or "torch._dynamo.optimize(" in frame.line:
+        if "eval_frame" in frame.filename or (frame.line is not None and "torch._dynamo.optimize(" in frame.line):
```

Comment:
How often is it the case that `frame.line` is `None`? I'm concerned because `self->frame->frame_obj` (in `THPPyInterpreterFrame_f_lineno`) is lazily instantiated.

Reply:
yeah I think you're right, in theory line could None according to this https://github.com/python/cpython/issues/94485 . In practice they probably should never be. 

Seems like frame_obj indeed got lazily initialized. Do you know how to trigger it to be created?

Reply:
We ran into this issue with pybind11. Accessing the attribute through the Python getattr API seems to trigger the lazy initialization and populates the value safely.

Reply:
@williamwen42 Accessing frame_obj is necessary here because Python std lib is asking for the line number from exception handlers which is only available from the frame object. 

> So we should not try to inspect frame_obj unless execution is complete.

I'm not sure I follow. The execution should be finished here already because there's already a Dynamo UserError thrown, and CPython is just trying to print out the line numbers from source files.

Reply:
"Execution is complete" meaning that CPython `_PyEval_EvalFrameDefault` is done with the frame. Error handling after this should be fine. I would suggest adding a test to make sure that for simple cases, the line number exists.